core/src/stored/dir_cmd.cc

```diff
@@ -920,10 +920,6 @@ static DeviceControlRecord* FindDevice(JobControlRecord* jcr,
                  device_resource->resource_name_, devname.c_str());
             continue;
           }
-          if (!device_resource->dev->autoselect) {
-            Dmsg1(100, "Device %s not autoselect skipped.\n", devname.c_str());
-            continue; /* device is not available */
-          }
           if (drive == kInvalidDriveNumber
```

Comment:
When you call `FindDevice()` with `drive == kInvalidDriveNumber` it will simply return the first drive (i.e. kInvalidDriveNumber means "I don't care which drive").
If the caller doesn't care, we should probably only return a device with `autoselect = yes`.
```suggestion
          if ((drive == kInvalidDriveNumber && device_resource->dev->autoselect)
```